app/controllers/api/v2/chapters_controller.rb

```diff
@@ -45,6 +45,24 @@ def changes
         render json: Api::V2::Changes::ChangeSerializer.new(@changes.changes, options).serializable_hash
       end
 
+      def headings
+        chapter_headings = chapter.headings
+
+        respond_to do |format|
+          format.csv do
+            send_data(
+              Api::V2::Csv::HeadingSerializer.new(chapter_headings).serialized_csv,
+              type: 'text/csv; charset=utf-8; header=present',
+              disposition: "attachment; filename=chapter-#{params[:id]}-headings-#{actual_date.iso8601}.csv",
+            )
+          end
+
+          format.any do
+            render json: Api::V2::Headings::HeadingSerializer.new(chapter_headings).serializable_hash
```

Comment:
Have you done any performance checks for this - my reading of this is

1. Load every heading under a Chapter
2. Serialize each heading
3. heading serializer loads _every_ commodity and compute `children?` for each one

In effect, heading serializer loads _every_ commodity under _every_heading for the chapter, and computes the `commodity.children?` for each one

This feels like like it will potentially take a long time to complete for some chapters - have you tried querying each of the chapters locally to see how long it takes to respond?

The alternative would be a cut down headings serializer for the chapters, which doesn't cascade into the commodities but not sure what was requested originally - there's nothing about this JSON endpoint in the ticket?

Reply:
This is wrong. For a heading to be a leaf we just need to work out whether there are any commodities not do stuff with every commodity

Reply:
Basically heading.commodities_dataset.count.positive?

Reply:
No, nothing in the ticket, that Json was not something required, so we can simplify the json, removing the commodities for example.
Not sure how to do it using the same serializer,  maybe using the option `:include`.

Reply:
Actually, I'm confusing myself. We should just remove this part. We don't need it

Reply:
Technically these apis are all just csv

Reply:
removed 👍